ionic-image-lazy-load.js

```diff
@@ -34,38 +35,71 @@ angular.module('ionicLazyLoad')
 
                 var deregistration = $scope.$on('lazyScrollEvent', function () {
                         //console.log('scroll');
                         if (isInView()) {
-                            
-                            //Bind "load" event
-                            $element.bind("load", function(e){
-                        		if($scope.lazyScrollResize == "true"){
-                        			//Call the resize to recalculate the size of the screen
-                        			$ionicScrollDelegate.resize();
-                        		}
-                        	});
-                            
-                            $element[0].src = $attributes.imageLazySrc; // set src attribute on element (it will load image)
+                            loadImage();
                             deregistration();
                         }
                     }
                 );
 
+                function loadImage() {
+                    //Bind "load" event
+                    $element.bind("load", function (e) {
+                        if ($attributes.imageLazyLoader) {
+                            loader.remove();
+                        }
+                        $element[0].src = $attributes.imageLazySrc; // set src attribute on element (it will load image)
+                        if ($scope.lazyScrollResize == "true") {
+                            //Call the resize to recalculate the size of the screen
+                            $ionicScrollDelegate.resize();
+                        }
+                    });
+
+                    if ($scope.imageLazyBackgroundImage == "true") {
+                        var bgImg = new Image();
+                        bgImg.onload = function () {
+                            if ($attributes.imageLazyLoader) {
+                                loader.remove();
+                            }
+                            $element[0].style.backgroundImage = 'url(' + $attributes.imageLazySrc + ')'; // set style attribute on element (it will load image)
+                            if ($scope.lazyScrollResize == "true") {
+                                //Call the resize to recalculate the size of the screen
+                                $ionicScrollDelegate.resize();
+                            }
+                        };
+                        bgImg.src = $attributes.imageLazySrc;
+                    }
+                }
+
                 function isInView() {
                     var clientHeight = $document[0].documentElement.clientHeight;
                     var clientWidth = $document[0].documentElement.clientWidth;
                     var imageRect = $element[0].getBoundingClientRect();
-                    return  (imageRect.top >= 0 && imageRect.bottom <= clientHeight) && (imageRect.left >= 0 && imageRect.right <= clientWidth);
+                    return (imageRect.top >= 0 && imageRect.top <= clientHeight + parseInt($attributes.imageLazyDistanceFromBottomToLoad))
```

Comment:
I changed imageRect.bottom to imageRect.top because I thought it was more natural to add a distance from the moment when the image start to appear on the screen.